hummingbot/market/hitbtc/hitbtc_active_order_tracker.pyx

```diff
@@ -0,0 +1,163 @@
+
+    @property
+    def active_asks(self) -> HitbtcOrderBookTrackingDictionary:
+        return self._active_asks
+
+    @property
+    def active_bids(self) -> HitbtcOrderBookTrackingDictionary:
+        return self._active_bids
+
+    def volume_for_ask_price(self, price) -> float:
+        return sum([float(msg["remaining_size"]) for msg in self._active_asks[price].values()])
+
+    def volume_for_bid_price(self, price) -> float:
+        return sum([float(msg["remaining_size"]) for msg in self._active_bids[price].values()])
+
+    def get_rates_and_quantities(self, entry) -> tuple:
+        return entry["R"], entry["Q"]
+
+    cdef tuple c_convert_diff_message_to_np_arrays(self, object message):
+        cdef:
+            dict content = message.content
+            list bid_entries = content["Z"]
+            list ask_entries = content["S"]
+            str order_id
+            str order_side
+            str price_raw
+            object price
+            dict order_dict
+            double timestamp = message.timestamp
+            double quantity = 0
+
+        bids = s_empty_diff
+        asks = s_empty_diff
+
+        if len(bid_entries) > 0:
+            bids = np.array(
+                [[timestamp,
+                  float(price),
+                  float(quantity),
+                  message.update_id]
+                 for price, quantity in [self.get_rates_and_quantities(entry) for entry in bid_entries]],
+                dtype="float64",
+                ndmin=2
+            )
+
+        if len(ask_entries) > 0:
+            asks = np.array(
+                [[timestamp,
+                  float(price),
+                  float(quantity),
+                  message.update_id]
+                 for price, quantity in [self.get_rates_and_quantities(entry) for entry in ask_entries]],
+                dtype="float64",
+                ndmin=2
+            )
+
+        return bids, asks
+
+    cdef tuple c_convert_snapshot_message_to_np_arrays(self, object message):
+        cdef:
+            object price
+            str order_id
+            str amount
+            dict order_dict
+
+        # Refresh all order tracking.
+        self._active_bids.clear()
+        self._active_asks.clear()
+        timestamp = message.timestamp
+
+        for snapshot_orders, active_orders in [(message.content["Z"], self._active_bids),
+                                              (message.content["S"], self.active_asks)]:
+
+            for order in snapshot_orders:
+                price = order["R"]
+                amount = str(order["Q"])
+                order_dict = {
+                    "order_id": timestamp,
+                    "quantity": amount
+                }
+
+                if price in active_orders:
+                    active_orders[price][timestamp] = order_dict
+                else:
+                    active_orders[price] = {
+                        timestamp: order_dict
+                    }
```

Comment:
I suspect the error with the conversion error, mentioned in the data source, has to do with this. 
You got to change how you access and process the `message.content`. Not every exchange have the same message format

Reply:
Looking at their API documentation, this should fix it:

<pre>
for snapshot_orders, active_orders in [(message.content["<b>bids</b>"], self._active_bids),
                                              (message.content["<b>asks</b>"], self.active_asks)]:

            for order in snapshot_orders:
                price = order["<b>price</b>"]
                amount = str(order["<b>size</b>"])
                order_dict = {
.
.
.
</pre>

Reply:
Thanks you very much, This should work.